src/main/java/ch/njol/skript/lang/VariableString.java

```diff
@@ -527,12 +520,50 @@ private static ChatColor getLastColor(CharSequence s) {
 	public String toString() {
 		return toString(null, false);
 	}
-	
+
+	/**
+	 * Parses all expressions in the string and returns it.
+	 * If this is a simple string, the event may be null.
+	 * 
+	 * @param e Event to pass to the expressions.
+	 * @return The input string with all expressions replaced.
+	 */
+	public String toString(@Nullable Event event) {
+		if (isSimple) {
+			assert simple != null;
+			return simple;
+		}
+		if (event == null)
+			throw new IllegalArgumentException("Event may not be null in non-simple VariableStrings!");
+
+		Object[] string = this.string;
+		assert string != null;
+		StringBuilder b = new StringBuilder();
+		List<Class<?>> types = new ArrayList<>();
+		for (Object o : string) {
+			if (o instanceof Expression<?>) {
+				Object[] objects = ((Expression<?>) o).getArray(event);
+				if (objects != null && objects.length > 0)
+					types.add(objects[0].getClass());
+				b.append(Classes.toString(objects, true, mode));
+			} else {
+				b.append(o);
+			}
+		}
+		String complete = b.toString();
+		if (!types.isEmpty()) {
+			DefaultVariables data = script.getData(DefaultVariables.class);
+			if (data != null)
+				data.add(complete, types.toArray(new Class<?>[0]));
+		}
```

Comment:
Is it intended that strings like `<red>Hello %player%!"` are added to the hints? This class is also used for regular strings in scripts

Reply:
Added a condition to ensure the user cannot define `<` and `>` in the variable structure.